config/webpack/webpack.config.js

```diff
@@ -106,7 +107,15 @@ const webpackConfig = (options, env, argv) => {
         test: /\.css$/,
         log: false,
         plugins: [
-          PostCSSCustomProperties({ preserve: true }),
+          PostCSSCustomProperties({
+            preserve: true,
+            // If we have a theme file, use the webpack promise to webpack it.  This promise will resolve to
+            // an object with themeable variables and values. This will then be used to update the end state CSS
+            // so that they are populated with values if variables aren't supported (e.g. IE10). This dance is
+            // necessary when code splitting to ensure the variables and values are applied across all code split
+            // css files
+            ...themeFile && { importFrom: [getThemeWebpackPromise(rootPath, themeFile)] },
```

Comment:
Can we run this wepback promise as a "prebuild" step instead of running webpack-within-webpack? 

We also need to manage what we do if the project errors out and providing meaningful error messaging so consumers know how to resolve this. For instance, if we dependency is missing, does this throw the same error messages as our config or what will that look like for consumers to start debugging?

Reply:
Unfortunately we can't do it like that.  the promise isn't synchronous and it needs to be guaranteed to happen before we run the main webpack.

Good call as far as errors, it should display the same errors as what webpack does by using the recommendation here: https://webpack.js.org/api/node/#error-handling

Reply:
https://github.com/cerner/terra-toolkit/pull/322/commits/b56514b05e791c048ac505b946d93f9c540ae8b7

Reply:
Could we wrap it in a promise chain to get around async? I saw @StephenEsser do something with auth recently. something like:

```
const themeFile = ThemeAggregator.aggregate();
const themePromise = themeFile ? getThemeWebpackPromise(rootPath, themeFile) : Promise.resolve(undefined);

return themePromise.then(themeOutput => webpackConfig({ ...options, themeOutput }, env, argv))

```